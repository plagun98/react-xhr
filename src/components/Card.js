import React from 'react';
import {Card, Col} from 'react-bootstrap';
import styled from 'styled-components';

const CardItem = props => {

    const CardI = styled(Card)`
        margin: 10px auto;
    `;
    const Image = styled(Card.Img)`
        max-height: 300px;
        width: auto;
    `;

    return (
        <Col md='4'>
            <CardI style={{ width: '18rem' }}>
                <Image variant="top" src={props.img} />
                <Card.Body>
                    <ul>
                        <li><span>Name:</span> {props.name}</li>
                        <li><span>Brand:</span> {props.brand}</li>
                        <li><span>Price:</span> {props.price} {props.currency}</li>
                        <li><span>Asin:</span> {props.asin}</li>
                        <li><span>Weight:</span> {props.weight}</li>
                    </ul>
                    <a href={props.link} target="_blank" rel="noopener noreferrer">Link</a>
                </Card.Body>
            </CardI>
        </Col>
    );
}

export default CardItem;