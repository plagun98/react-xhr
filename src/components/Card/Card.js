import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import { Card, Col } from 'react-bootstrap';

const CardI = styled(Card)`
    margin: 10px auto;
`;
const Image = styled(Card.Img)`
    max-height: 300px;
    width: auto;
`;
const ColStyled = styled(Col)`
    padding: 0;

`;

const CardItem = props => (

    <ColStyled md='4'>
        <CardI style={{ width: '18rem' }}>
            <Image variant="top" src={props.img} />
            <Card.Body>
                <ul>
                    <li><span>Name:</span> {props.name}</li>
                    <li><span>Brand:</span> {props.brand}</li>
                    <li><span>Price:</span> {props.price} {props.currency}</li>
                    <li><span>Asin:</span> {props.asin}</li>
                    <li><span>Weight:</span> {props.weight}</li>
                </ul>
                <a href={props.link} target="_blank" rel="noopener noreferrer">Link</a>
            </Card.Body>
        </CardI>
    </ColStyled>
);

CardItem.propTypes = {
    name: PropTypes.string,
    brand: PropTypes.string,
    price: PropTypes.number,
    currency: PropTypes.string,
    asin: PropTypes.string,
    weight: PropTypes.number,
    img: PropTypes.string,
    link: PropTypes.string
}

export default CardItem;