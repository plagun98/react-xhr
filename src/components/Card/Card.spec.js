import Card from './Card';

it('render test', () => {
    expect(Card).toMatchSnapshot()
});