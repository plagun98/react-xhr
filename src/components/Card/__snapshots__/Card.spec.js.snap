// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render test 1`] = `[Function]`;
